File: lib/native_admob_options.dart
import 'package:flutter/material.dart';

class NativeTextStyle {
  final double? fontSize;
  final Color? color;
  final Color? backgroundColor;
  final bool isVisible;

  const NativeTextStyle({
    this.fontSize,
    this.color,
    this.backgroundColor,
    this.isVisible = true,
  });

  Map<String, dynamic> toJson() => {
        "backgroundColor": backgroundColor != null ? "#${backgroundColor!.value.toRadixString(16)}" : null,
        "fontSize": fontSize,
        "color": color != null ? "#${color!.value.toRadixString(16)}" : null,
        "isVisible": isVisible,
      };
}

class NativeAdmobOptions {
  final bool showMediaContent;
  final Color ratingColor;
  final Color? backgroundColor;
  final NativeTextStyle adLabelTextStyle;
  final NativeTextStyle headlineTextStyle;
  final NativeTextStyle advertiserTextStyle;
  final NativeTextStyle bodyTextStyle;
  final NativeTextStyle storeTextStyle;
  final NativeTextStyle priceTextStyle;
  final NativeTextStyle callToActionStyle;

  const NativeAdmobOptions({
    this.showMediaContent = true,
    this.ratingColor = Colors.yellow,
    this.backgroundColor,
    this.adLabelTextStyle = const NativeTextStyle(
      fontSize: 12,
      color: Colors.white,
      backgroundColor: Color(0xFFFFCC66),
    ),
    this.headlineTextStyle = const NativeTextStyle(
      fontSize: 16,
      color: Colors.black,
    ),
    this.advertiserTextStyle = const NativeTextStyle(
      fontSize: 14,
      color: Colors.black,
    ),
    this.bodyTextStyle = const NativeTextStyle(
      fontSize: 12,
      color: Colors.grey,
    ),
    this.storeTextStyle = const NativeTextStyle(
      fontSize: 12,
      color: Colors.black,
    ),
    this.priceTextStyle = const NativeTextStyle(
      fontSize: 12,
      color: Colors.black,
    ),
    this.callToActionStyle = const NativeTextStyle(
      fontSize: 15,
      color: Colors.white,
      backgroundColor: Color(0xFF4CBE99),
    ),
  });

  Map<String, dynamic> toJson() => {
        "showMediaContent": this.showMediaContent,
        "ratingColor": "#${ratingColor.value.toRadixString(16)}",
        "backgroundColor": backgroundColor != null ? "#${backgroundColor!.value.toRadixString(16)}" : null,
        "adLabelTextStyle": adLabelTextStyle.toJson(),
        "headlineTextStyle": headlineTextStyle.toJson(),
        "advertiserTextStyle": advertiserTextStyle.toJson(),
        "bodyTextStyle": bodyTextStyle.toJson(),
        "storeTextStyle": storeTextStyle.toJson(),
        "priceTextStyle": priceTextStyle.toJson(),
        "callToActionStyle": callToActionStyle.toJson(),
      };
}

File: ios/Classes/NativeAdViews/NativeAdmobOptions.swift
//
//  BannerOptions.swift
//  flutter_native_admob
//
//  Created by Dao Duy Duong on 3/8/20.
//

import UIKit

struct NativeAdmobOptions {
    
    struct NativeTextStyle {
        var fontSize: CGFloat
        var color: UIColor
        var backgroundColor: UIColor? = nil
        var isVisible: Bool = true
        var iosTypeface: String? = nil
        
        mutating func update(JSON: [String: Any]) {
            if let fontSize = JSON["fontSize"] as? CGFloat {
                self.fontSize = fontSize
            }
            
            if let colorString = JSON["color"] as? String {
                self.color = .fromHex(colorString)
            }
            
            if let backgroundColorString = JSON["backgroundColor"] as? String {
                self.backgroundColor = .fromHex(backgroundColorString)
            }
            
            if let isVisible = JSON["isVisible"] as? Bool {
                self.isVisible = isVisible
            }
            
            if let iosTypeface = JSON["iosTypeface"] as? String {
                self.iosTypeface = iosTypeface
            }
        }
    }
    
    var showMediaContent: Bool = true
    var backgroundColor: UIColor?
    var ratingColor: UIColor = .yellow

    var adLabelTextStyle = NativeTextStyle(fontSize: 12, color: .white, backgroundColor: .fromHex("FFCC66"))
    var headlineTextStyle = NativeTextStyle(fontSize: 16, color: .black)
    var advertiserTextStyle = NativeTextStyle(fontSize: 14, color: .black)
    var bodyTextStyle = NativeTextStyle(fontSize: 12, color: .gray)
    var storeTextStyle = NativeTextStyle(fontSize: 12, color: .black)
    var priceTextStyle = NativeTextStyle(fontSize: 12, color: .black)
    var callToActionStyle = NativeTextStyle(fontSize: 15, color: .white, backgroundColor: .fromHex("#4CBE99"))
    
    init() {}
    
    init(_ data: [String: Any]) {
        if let showMediaContent = data["showMediaContent"] as? Bool {
            self.showMediaContent = showMediaContent
        }


        if let backgroundColorString = data["backgroundColor"] as? String {
            backgroundColor = .fromHex(backgroundColorString)
        }
        
        if let ratingColorString = data["ratingColor"] as? String {
            ratingColor = .fromHex(ratingColorString)
        }
        
        if let data = data["adLabelTextStyle"] as? [String: Any] {
            adLabelTextStyle.update(JSON: data)
        }
        
        if let data = data["headlineTextStyle"] as? [String: Any] {
            headlineTextStyle.update(JSON: data)
        }
        
        if let data = data["advertiserTextStyle"] as? [String: Any] {
            advertiserTextStyle.update(JSON: data)
        }
        
        if let data = data["bodyTextStyle"] as? [String: Any] {
            bodyTextStyle.update(JSON: data)
        }
        
        if let data = data["storeTextStyle"] as? [String: Any] {
            storeTextStyle.update(JSON: data)
        }
        
        if let data = data["priceTextStyle"] as? [String: Any] {
            priceTextStyle.update(JSON: data)
        }
        
        if let data = data["callToActionStyle"] as? [String: Any] {
            callToActionStyle.update(JSON: data)
        }
    }
}

